### ISSUE TITLE
cr-app-elk771 - Init very slow

### ISSUE THREAD

**Issue by GabboPenna** [bluek8s/kubedirector#427]
Hi, how come in the init phase of the pods a series of folders are copied?

`/bin/bash -c ! [ -f /mnt/etc/kubedirector.init ] && cp --parent -ax /etc /opt /usr /var /mnt; touch /mnt/etc/kubedirector.init`

This slows down the deployment phase a lot.
A solution to speed up?

**Comment by joel-bluedata** [comment]
Right, this copy is only done if persistent storage is configured. For directories that need to have changes persist across pod restarts, those directories need to be mounted on persistent storage. However if those directories also have initial content in them (in the container image), that content needs to be copied to the persistent storage before we can do such a mount.

If there's a LOT of copying happening, slowing down the initial deployment, then one or more of these two things is true:

* The container image is laid out so that "places where user data is written/modified" are intermixed with other nonmutable data, and so it's difficult to arrange directory mounts that only affect that user data.
* Or, the mount directories currently specified in the kdapp are too broad, and should be refined (perhaps into certain subdirectories) so that they more exactly cover only the user data locations and don't cover as much nonmutable data.

I don't personally know which of these is the case for the ELK kdapp; we'll need to follow up with the kdapp author.

**Comment by joel-bluedata** [comment]
Saurabh, I'm parking this on with you for now since Om Katthar doesn't have an active account here now. If you don't know who the right person to tackle this is, we will find out.

**Comment by swamibluedata** [comment]
Should we change our appConfigDefaultMountFolders to be selective "/opt/guestconfig" instead of "/opt"

Also about "/usr". Limit to set of binaries that we require?

**Comment by joel-bluedata** [comment]
We should be as specific as we can for any persisted directories yeah. Unfortunately changing the defaultPersistDirs will probably require updates in various apps and is in general a non-backwards-compatible change, but this problem is coming up a lot.

**Comment by joel-bluedata** [comment]
We should see now whether cr-app-elk771 can be moved to the new storage layout.
